### PR TITLE
fix-json-tag

### LINKED ISSUE #5: `CreatePage` function always return 400 error
First of all ,thanks for creating the notino-sdk for go. 

when the following code execute always return 400 error. 

`go version go1.15.6 darwin/amd64`

```
func getClient(apiKey string) *notion.Client {
	return notion.NewClient(apiKey)
}

func createPage(c echo.Context) error {
	notionApi := os.Getenv("NOTION_API")
	client    := getClient(notionApi)
	ctx       := context.Background()
	parentId  := "{page_id}"

	titleText := &notion.Text{
		Content: "this is a title of the created page",
	}
	title := []notion.RichText{
		{
			Type: notion.RichTextTypeText,
			Text: titleText,
		},
	}
	params := notion.CreatePageParams{
		ParentType: notion.ParentTypePage,
		ParentID:   parentId,
		Title:      title,
	}

	page, err := client.CreatePage(ctx, params)
	if err != nil {
		fmt.Printf("%s",err)
	}
        fmt.Println(page)
	return nil
}
```

The following error code will be returned when `createPage` function is executed.

```
notion: failed to create page: body failed validation. Fix one: body.parent.type should be not present, instead was `"database_id"`. body.parent.page_id should be defined, instead was `undefined`. (code: validation_error, status: 400)
```

I think the structure is correct, so I'm thinking there's probably something wrong with the code.

Thank you .

### PR DESCRIPTION
**Pull request by kajirita2002**
Hi!!!

fix #5 

Hit the api by curl command as follows

```sh
curl -X POST https://api.notion.com/v1/pages \
         -H "Authorization: Bearer XXXX" \
         -H "Content-Type: application/json" \
         -H "Notion-Version: 2021-05-13" \
         --data '{
           "parent": { 
               "type":"database_id",
               "database_id":"f38a21302e234460b6e7b0682bad2428"},
               "properties":{
                   "Name":{
                       "type":"",
                       "title":[{
                           "type":"","text":{"content":"hello world"}}
                      ]
                    }
                } 
            }
        }'
        
```
This happen 

```sh
{"object":"error","status":401,"code":"unauthorized","message":"API token is invalid."}
```

so, I fix it

### CODE REVIEW

**Comment by dstotijn** [comment]
Instead of duplicating the table test, could you add a _new_ test case (e.g. with name "database parent, successful response") where you assert that the `POST` body doesn't contain `title: ""`? The response data coming from Notion will contain the `title` key, so we can leave it in the _response_. Thanks!

**Comment by dstotijn** [comment]
FYI, ran into the empty `type` key issue as well while I was writing tests for a different endpoint ("update page properties"). So I made the `omitempty` change in [this commit](https://github.com/dstotijn/go-notion/commit/ee6b002819d6d68ea2a5fa8bdfd660ab73c10293) already. But if you want we can still use this PR to update the tests for the "create page" endpoint. 👍

**Comment by kajirita2002** [comment]
Ok! I want to update it because this is my first PR to OSS!!

**Comment by kajirita2002** [comment]
I fix this now. please check it.
https://github.com/dstotijn/go-notion/pull/7/commits/40deeda293bc3071b1e4aa72dd5dd0b15fdc9453

File: client_test.go [hunk 1]
```diff
@@ -907,6 +907,130 @@ func TestCreatePage(t *testing.T) {
 			},
 			expError: nil,
 		},
+		{
+			name: "database parent, successful response",
+			params: notion.CreatePageParams{
+				ParentType: notion.ParentTypePage,
```
Comment (dstotijn) [inline client_test.go:913]: Can you change this table test to use `ParentType: notion.ParentTypeDatabase`? The idea is that we want to assert that when you create a _database_ page, that the `type` of DatabasePageProperty's are not included in the JSON `POST` body. Right now, the test is using a _parent_ page.

**Review by dstotijn — CHANGES_REQUESTED** [review]

File: client_test.go (hunk 1 above)
Comment (kajirita2002) [reply client_test.go:913]: OK! I will👍

**Comment by kajirita2002** [comment]
I fix like this https://github.com/dstotijn/go-notion/pull/7/commits/d2010129197f1cfd484baacab79f02da91a4f9cf
But, test cannot pass... I could not understand not that test cannot pass.
Do you know this solution?

**Comment by sivchari** [comment]
I fix this PR in #10
Please check it 👍

**Comment by kajirita2002** [comment]
@sivchari thanks🤣

**Comment by dstotijn** [comment]
Further finished up in #7

**Pull request closed without merging**